include/alpaka/mem/buf/cpu/Fill.hpp

```diff
@@ -0,0 +1,146 @@
+            Vec<TDim, DstSize> const m_dstExtent;
+#endif
+            Vec<TDim, DstSize> const m_dstPitchBytes;
+            std::uint8_t* const m_dstMemNative;
+        };
+
+        // Generic ND version
+        template<typename TDim, typename TView, typename TExtent, typename TValue>
+        struct TaskFillCpu : public TaskFillCpuBase<TDim, TView, TExtent, TValue>
+        {
+            using Base = TaskFillCpuBase<TDim, TView, TExtent, TValue>;
+            using Base::Base;
+            using Elem = typename Base::Elem;
+            using ExtentSize = typename Base::ExtentSize;
+
+            // using DstSize = typename Base::DstSize;
+            // using DimMin1 = DimInt<TDim::value - 1u>;
+
+            ALPAKA_FN_HOST auto operator()() const -> void
+            {
+                // Vec<DimMin1, ExtentSize> const extentWithoutInnermost = subVecBegin<DimMin1>(this->m_extent);
+                // Vec<DimMin1, DstSize> const pitchBytesWithoutOutmost = subVecBegin<DimMin1>(this->m_dstPitchBytes);
+
+                if(static_cast<std::size_t>(this->m_extent.prod()) != 0u)
+                {
+                    meta::ndLoopIncIdx(
+                        this->m_extent,
+                        [&](Vec<TDim, ExtentSize> const& idx)
+                        {
+                            std::uintptr_t offsetBytes
+                                = static_cast<std::uintptr_t>((idx * this->m_dstPitchBytes).sum());
+                            assert(offsetBytes % alignof(Elem) == 0);
+                            Elem* elem = reinterpret_cast<Elem*>(
+                                __builtin_assume_aligned(this->m_dstMemNative + offsetBytes, alignof(Elem)));
+
+                            *elem = this->m_value;
+                        });
+                }
+            }
+        };
+
+        // 0D version (scalar fill)
+        template<typename TView, typename TExtent, typename TValue>
+        struct TaskFillCpu<DimInt<0u>, TView, TExtent, TValue>
+        {
+            using Elem = alpaka::Elem<TView>;
+
+            static_assert(std::is_same_v<Elem, TValue>, "Fill value must match view element type");
+
+            template<typename TViewFwd>
+            TaskFillCpu(TViewFwd&& view, TValue const& value, [[maybe_unused]] TExtent const& extent)
+                : m_value(value)
+                , m_dstMemNative(reinterpret_cast<std::uint8_t*>(getPtrNative(view)))
+            {
+                ALPAKA_ASSERT(getExtents(extent).prod() == 1u);
+                ALPAKA_ASSERT(getExtents(view).prod() == 1u);
+                ALPAKA_ASSERT(reinterpret_cast<std::uintptr_t>(m_dstMemNative) % alignof(Elem) == 0);
+            }
+
+            ALPAKA_FN_HOST auto operator()() const noexcept -> void
+            {
+                *reinterpret_cast<Elem*>(m_dstMemNative) = m_value;
+            }
+
+            TValue const m_value;
+            std::uint8_t* const m_dstMemNative;
```

Comment:
For the 0-dimensional case there cannot be any padding involved, so we can simplify the code:
  - store the address as an `Elem*`
  - avoid the cast to and from `uint8_t*`